src/chia_log/log_consumer.py

```diff
@@ -107,32 +106,42 @@ def stop(self):
         self._is_running = False
 
     def _consume_loop(self):
-        logging.info(f"Consuming remote log file {self._remote_log_path} from {self._remote_host} ({self._remote_platform})")
+        logging.info(
+            f"Consuming remote log file {self._remote_log_path} from {self._remote_host} ({self._remote_platform})"
+        )
 
         if self._remote_platform == OS.WINDOWS:
-            stdin, stdout, stderr = self._ssh_client.exec_command(f"powershell.exe Get-Content {self._remote_log_path} -Wait -Tail 1")
+            stdin, stdout, stderr = self._ssh_client.exec_command(
+                f"powershell.exe Get-Content {self._remote_log_path} -Wait -Tail 1"
+            )
         else:
             stdin, stdout, stderr = self._ssh_client.exec_command(f"tail -F {self._remote_log_path}")
 
         while self._is_running:
             log_line = stdout.readline()
             self._notify_subscribers(log_line)
 
-    def _get_remote_platform(self):
-        stdin, stdout, stderr = self._ssh_client.exec_command(f"uname -a")
-        fout: str = stdout.readline().lower()
-        ferr: str = stderr.readline().lower()
-
-        if 'linux' in fout:
-            return OS.LINUX
-        elif 'darwin' in fout:
-            return OS.MACOS
-        elif 'not recognized' in ferr:
-            return OS.WINDOWS
-        else:
-            logging.error(f"Found unsupported platform on remote host, assuming Linux and hope for the best.")
 
-        return OS.LINUX
+def get_host_info(host: str, user: str, path: str):
+
+    client = paramiko.client.SSHClient()
+    client.load_system_host_keys()
+    client.connect(hostname=host, username=user)
+
+    stdin, stdout, stderr = client.exec_command("uname -a")
+    fout: str = stdout.readline().lower()
+    ferr: str = stderr.readline().lower()
+
+    if "linux" in fout:
+        return OS.LINUX, PurePosixPath(path)
+    elif "darwin" in fout:
+        return OS.MACOS, PurePosixPath(path)
+    elif "not recognized" in ferr:
+        return OS.WINDOWS, PureWindowsPath(path)
+    else:
+        logging.error("Found unsupported platform on remote host, assuming Linux and hope for the best.")
+
+    return OS.LINUX
```

Comment:
If you have specify the return type annotation as I suggest and run the [formatting, type-checking and linting](https://github.com/martomi/chiadog/blob/main/CONTRIBUTING.md#formatting-and-linting), it should automatically detect the wrong return here :-)